objective-c/route_guide/README.md

```diff
@@ -0,0 +1,280 @@
+```protobuf
+  // Accepts a stream of RouteNotes sent while a route is being traversed,
+  // while receiving other RouteNotes (e.g. from other users).
+  rpc RouteChat(stream RouteNote) returns (stream RouteNote) {}
+```
+
+Our .proto file also contains protocol buffer message type definitions for all the request and response types used in our service methods - for example, here's the `Point` message type:
+```protobuf
+// Points are represented as latitude-longitude pairs in the E7 representation
+// (degrees multiplied by 10**7 and rounded to the nearest integer).
+// Latitudes should be in the range +/- 90 degrees and longitude should be in
+// the range +/- 180 degrees (inclusive).
+message Point {
+  int32 latitude = 1;
+  int32 longitude = 2;
+}
+```
+
+You can specify a prefix to be used for your generated classes by adding the `objc_class_prefix` option at the top of the file. For example:
+```protobuf
+option objc_class_prefix = "RTG";
+```
+
+
+<a name="protoc"></a>
+## Generating client code
+
+Next we need to generate the gRPC client interfaces from our .proto service definition. We do this using the protocol buffer compiler (`protoc`) with a special gRPC Objective-C plugin.
+
+For simplicity, we've provided a [Podspec file](https://github.com/grpc/grpc-common/blob/master/objective-c/route_guide/RouteGuide.podspec) that runs `protoc` for you with the appropriate plugin, input, and output, and describes how to compile the generated files. You just need to run in this directory:
+
+```shell
+$ pod install
+```
+
+which, before installing the generated library in the XCode project of this sample, runs:
+
+```shell
+$ protoc -I ../../protos --objc_out=Pods/RouteGuide --objcgrpc_out=Pods/RouteGuide ../../protos/route_guide.proto
+```
+
+Running this command generates the following files in under `Pods/RouteGuide/`:
+- `RouteGuide.pbobjc.h`, the header which declares your generated message classes.
+- `RouteGuide.pbobjc.m`, which contains the implementation of your message classes.
+- `RouteGuide.pbrpc.h`, the header which declares your generated service classes.
+- `RouteGuide.pbrpc.m`, which contains the implementation of your service classes.
+
+These contain:
+- All the protocol buffer code to populate, serialize, and retrieve our request and response message types.
+- A class called `RTGRouteGuide` that for clients to call with the methods defined in the `RouteGuide` service.
+
+The provided Podspec file works for any proto library you define; you just need to replace the name (matching the file name), version, and other metadata.
```

Comment:
Slight rephrase:

You can also use the provided Podspec file to generate client code from any other proto service definition; just replace the name (matching the file name), version, and other metadata.